compiler/resolution/tuples.cpp

```diff
@@ -799,13 +800,14 @@ do_computeTupleWithIntent(bool valueOnly, IntentTag intent, Type* t)
 {
   INT_ASSERT(t->symbol->hasFlag(FLAG_TUPLE));
 
-  FnSymbol*  typeConstructor    = t->defaultTypeConstructor;
-  BlockStmt* instantiationPoint = typeConstructor->instantiationPoint;
-
   // Construct tuple that would be used for a particular argument intent.
 
   bool allSame = true;
   AggregateType* at = toAggregateType(t);
+
+  FnSymbol*  typeConstructor    = at->defaultTypeConstructor;
```

Comment:
I think lines 806 and 808 imply that `t' must also be an AggregateType.  Does that seem right to you?
If so I'd be inclined to update the function signature accordingly.